.github/workflows/deploy.yml

```diff
@@ -0,0 +1,45 @@
+name: Deploy
+
+on: push
+
+jobs:
+  deploy:
+    if: github.repository_owner == 'meniga'
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0 
+      - uses: actions/cache@v2
+        with:
+          path: |
+            ~/.gradle/caches
+            ~/.gradle/wrapper
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
+      - name: Set environment variables
+        run: |
+          echo "DEPLOY_BUILD_NUMBER=$((500 + ${{ github.run_number }}))" >> $GITHUB_ENV
+      - name: Deploy branch snapshot
+        if: "!startsWith(github.ref, 'refs/tags/')"
+        env:
+            BINTRAY_CREDENTIALS: ${{ secrets.BINTRAY_CREDENTIALS}}
```

Comment:
just a minor idea @WezSieTato - maybe we can keep those variables at the job level instead of steps? This was they wouldn't be declared twice